### PR TITLE
4.x Alerts

### LINKED ISSUE #770: 4.x - add alerts and new alert variants
The current alerts only have one design option: https://uids.brand.uiowa.edu/components/detail/alert.html.   In the 4.x branch, it would be ideal to have an alert/message component that has left and center variants that can be more flexible for use as:

- Alert banners for displaying a prominent message at the top of the screen.
- A section message is used to alert users to a particular section of the screen.
- Inline message to let users know when important information is available or when an action is required.

# Proposed solution
Move over the current alerts component from UIDS 3.x and create new variants/stories for the uses listed below. 
  - Hawk alert left aligned, center aligned.
  - Inline alert, outlined, left aligned, center aligned.
-  Use Font Awesome stacking styles for icon: 

```
<span class="fa-stack fa-1x"><span class="fas fa-circle fa-stack-2x"></span><span class="fa-link fas fa-stack-1x fa-inverse"></span></span>
``` 

## Hawk Alert

**Left aligned:**

Screen Shot 2022-09-15 at 9 56 44 AM

**Centered:** TBD

## Section message

**Left aligned:**

Screen Shot 2022-09-15 at 8 40 19 AM
Screen Shot 2022-09-15 at 8 39 47 AM
Screen Shot 2022-09-15 at 8 39 32 AM
Screen Shot 2022-09-15 at 8 39 02 AM
Screen Shot 2022-09-15 at 8 38 03 AM

**Centered:** TBD

## Inline message

**Left aligned:**

Screen Shot 2022-09-14 at 3 37 50 PM
Screen Shot 2022-09-15 at 8 48 08 AM
Screen Shot 2022-09-15 at 8 47 24 AM

**Centered:** TBD

### PR DESCRIPTION
**Pull request by e-marie-w**
Resolves #770.

### CODE REVIEW

**Comment by bspeare** [comment]
These are all resolved now

<details>

Noticing the sizing difference between `Lorem ipsum sit dolor amet.<p>Lorem ipsum sit dolor amet.</p>` can we get the text without `<p>` to render the same as the text with `<p>` markup?  

Also, can we increase the icon size to like `.alert .fa-1x {font-size: 1.5em;}`

With the `<p>` having extra bottom margin, the vertically centered icon will need to be evaluated. 

Screenshot 2023-08-17 at 9 31 37 AM

Same issue with just a paragraph:

Screenshot 2023-08-17 at 9 34 35 AM

Could be solved by `.alert p + p {margin-top: 1.05rem;}` type rule. 
 
Update: This will also need to be done for heading tags as well. 

</details>

**Comment by bspeare** [comment]
I'm finished with my review.  This looks great to me and I will leave this open in case others want to review.

File: src/assets/scss/_variables.scss [hunk 1]
```diff
@@ -19,6 +19,8 @@ $warning-light: lighten($warning, 45%);
 $danger: #BD472A;
 $danger-light: lighten($danger, 50%);
 
+$orange: #BD472A;
+
```
Comment (pyrello) [inline src/assets/scss/_variables.scss]: Please move this to be grouped with the other colors above the alert colors.

File: src/components/alert/Alert.stories.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  },
+  // And then the `args` are bound to your component with `v-bind="args"`
+  template: `
+    <uids-alert
+      :type="args.type"
+      :centered="args.centered"
+      :dismissible="args.dismissible"
+      :centerIconVertically="args.centerIconVertically"
+    >
+      <template #default><div v-html="args.default"></div></template>
```
Comment (pyrello) [inline src/components/alert/Alert.stories.js]:
```suggestion
      <div v-html="args.default"></div>
```
I don't think this is necessary when just using the default slot.

File: src/components/alert/Alert.stories.js [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+Info.args = {
+  type: 'info',
+  centered: false,
+  dismissible: false,
+  centerIconVertically: false,
+  default: '' +
+    '<h2 class="headline headline--serif">' +
+      'Alert title' +
+    '</h2>' +
+    '<p>Lorem ipsum sit dolor amet.</p>',
```
Comment (pyrello) [inline src/components/alert/Alert.stories.js]:
```suggestion
  default: `
    <h2 class="headline headline--serif">
      Alert title
    </h2>
    <p>Lorem ipsum sit dolor amet.</p>
`,
```

File: src/components/alert/Alert.vue [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+  }
+  else if (props.type == 'warning') {
+    return 'exclamation-triangle';
+  }
+  else if (props.type == 'danger') {
+    return 'exclamation';
+  }
+  else {
+    return 'info';
+  }
```
Comment (pyrello) [inline src/components/alert/Alert.vue]:
```suggestion
  switch (props.type) {
    case 'success':
      return 'check';
      break;
    case 'warning':
      return 'exclamation-triangle';
      break;
    case 'danger':
      return 'exclamation';
      break;
    default:
      return 'info';
      break;
  }
```

File: src/components/alert/Alert.stories.js [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+      name: 'Content',
+      control: { type: 'text' },
+      table: {
+        category: 'Content',
+      },
+    },
+  },
+};
+
+// More on component templates: https://storybook.js.org/docs/vue/writing-stories/introduction#using-args
```
Comment (pyrello) [inline src/components/alert/Alert.stories.js]:
```suggestion
```
Let's remove this for actual components.

File: src/components/alert/Alert.stories.js [hunk 6]
```diff
@@ -0,0 +1,103 @@
+import UidsAlert from './Alert.vue';
+
+// More on default export: https://storybook.js.org/docs/vue/writing-stories/introduction#default-export
```
Comment (pyrello) [inline src/components/alert/Alert.stories.js]:
```suggestion
```

Removing comment for actual component.

File: src/components/alert/Alert.stories.js [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+        category: 'Content',
+      },
+    },
+  },
+};
+
+const Template = (args) => ({
+  // Components used in your story `template` are defined in the `components` object
+  components: { UidsAlert },
+  // The story's `args` need to be mapped into the template through the `setup()` method
```
Comment (pyrello) [inline src/components/alert/Alert.stories.js]:
```suggestion
```

File: src/components/alert/Alert.stories.js [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+      control: { type: 'text' },
+      table: {
+        category: 'Content',
+      },
+    },
+  },
+};
+
+const Template = (args) => ({
+  // Components used in your story `template` are defined in the `components` object
```
Comment (pyrello) [inline src/components/alert/Alert.stories.js]:
```suggestion
```

File: src/components/alert/Alert.stories.js [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+  components: { UidsAlert },
+  // The story's `args` need to be mapped into the template through the `setup()` method
+  setup() {
+    return { args };
+  },
+  // And then the `args` are bound to your component with `v-bind="args"`
```
Comment (pyrello) [inline src/components/alert/Alert.stories.js]:
```suggestion
```

File: src/components/alert/Alert.stories.js [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+      :centered="args.centered"
+      :dismissible="args.dismissible"
+      :centerIconVertically="args.centerIconVertically"
+    >
+      <div v-html="args.default"></div>
+    </uids-alert>`,
+})
+
+export const Info = Template.bind({});
+// More on args: https://storybook.js.org/docs/vue/writing-stories/args
```
Comment (pyrello) [inline src/components/alert/Alert.stories.js]:
```suggestion
```

File: src/components/alert/Alert.stories.js [hunk 11]
```diff
@@ -0,0 +1,98 @@
+import UidsAlert from './Alert.vue';
+
+export default {
+  title: 'Components/Alert',
+  component: UidsAlert,
+  // More on argTypes: https://storybook.js.org/docs/vue/api/argtypes
```
Comment (pyrello) [inline src/components/alert/Alert.stories.js]:
```suggestion
```

**Review by pyrello — APPROVED** [review]
Code review looks good.

**Pull request merged**